carfield.mk

```diff
@@ -209,12 +210,16 @@ pulpd-sw-build: pulpd-sw-init
 # Generate HW #
 ###############
 
+include $(IDMA_ROOT)/idma.mk
+
 ## @section Carfield platform HW generation
 .PHONY: car-hw-init
 ## Initialize Carfield HW. This step takes care of the generation of the missing hardware or the
 ## update of default HW configurations in some of the domains. See the two prerequisite's comment
 ## for more information.
-car-hw-init: spatzd-hw-init chs-hw-init secd-hw-init
+
+car-hw-init: idma_hw_all spatzd-hw-init chs-hw-init secd-hw-init
```

Comment:
I would prefer to have these into Cheshire (I think Chao already did it [here](https://github.com/pulp-platform/cheshire/commit/4f6c9d4a74b79db350568e56ab153888eb14ac7c#diff-aa23389e058ef9c90c8225adbbe8c32342bdda6fdb0730f6567e19d22ace0fb7L120-L154))

Reply:
if uncomment [this](https://github.com/pulp-platform/cheshire/commit/945d0a6e34437d35fedfab7c923bbab62d89433c#diff-aa23389e058ef9c90c8225adbbe8c32342bdda6fdb0730f6567e19d22ace0fb7R123) then should be ok, otherwise the rtl is not generated. I don't think cheshire could compile without running that command.

Reply:
no sorry, uncommenting is not enought, the commented target shold be added as dependency like this:
`#include $(IDMA_ROOT)/idma.mk
#$(IDMA_ROOT)/.generated: idma_hw_all
$(IDMA_ROOT)/.generated: idma_hw_all $(IDMA_ROOT)/target/rtl/idma_reg64_2d.hjson
	flock -x $@ sh -c "cp $< $(dir $@)/target/rtl/; $(MAKE) -j1 otp" && touch $@`

Reply:
I think Chao added the rule to the `CHS_HW_ALL` so it should work automatically, but I will double check.

Reply:
Update: she added the rule and commented it out, but can be enabled directly into Cheshire (needs flatdict).